pandas/sparse/array.py

```diff
@@ -769,14 +769,17 @@ def make_sparse(arr, kind='block', fill_value=None):
     if isnull(fill_value):
         mask = notnull(arr)
     else:
-        mask = arr != fill_value
+        mask = np.not_equal(arr, fill_value)
+        # In NumPy 1.12.0, not implemented at least for arr.dtype=str
+        if mask is NotImplemented:
+            mask = np.not_equal(arr.astype(object), fill_value)
```

Comment:
Well, `mask = arr != fill_value` test was there before since 2011. And this test fails when `arr.dtype.kind == 'U' and type(fill_value) != str` (i.e. comparing a vector of fixed-length strings to a non-string). Or at least `mask` gets set to _scalar_ `False` instead of a vector of boolean values, which then fails a few lines later on. It breaks on [this test line](https://github.com/pandas-dev/pandas/pull/15497/files#diff-b30cded911d05ab8f4b2ab86c7dcab58R1169), which, in my opinon, should generally pass.